service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -213,39 +218,66 @@ public void send(String deviceId, String moduleId, Message message) throws IOExc
 
     /**
      * Get FeedbackReceiver object.This API has been deprecated. Use new API without deviceId as an input parameter.
-     * @deprecated As of release 1.1.15, replaced by {@link #getFeedbackReceiver()}
+     * @deprecated Use {@link #createFeedbackMessageListener(FeedbackBatchMessageCallback)} instead
      * @param deviceId The device identifier for the target device
      * @return The instance of the FeedbackReceiver
      */
-    @Deprecated public FeedbackReceiver getFeedbackReceiver(String deviceId)
+    @Deprecated
+    public FeedbackReceiver getFeedbackReceiver(String deviceId)
     {
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_017: [The function shall create a FeedbackReceiver object and returns with it. This API is deprecated.]
-        FeedbackReceiver feedbackReceiver = new FeedbackReceiver(hostName, userName, sasToken, iotHubServiceClientProtocol, deviceId);
-        return feedbackReceiver;
+        return new FeedbackReceiver(hostName, userName, makeSasToken(), iotHubServiceClientProtocol, deviceId);
     }
     
      /**
      * Get FeedbackReceiver object.  
-     *
-     *
+     * @deprecated Use {@link #createFeedbackMessageListener(FeedbackBatchMessageCallback)} instead
      * @return The instance of the FeedbackReceiver
      */
-    
-     public FeedbackReceiver getFeedbackReceiver()
+    @Deprecated
+    public FeedbackReceiver getFeedbackReceiver()
     {
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_018: [The function shall create a FeedbackReceiver object and returns with it. This API doesn't need deviceId as an input parameter]
-        FeedbackReceiver feedbackReceiver = new FeedbackReceiver(hostName, userName, sasToken, iotHubServiceClientProtocol);
-        return feedbackReceiver;
+        return new FeedbackReceiver(hostName, userName, makeSasToken(), iotHubServiceClientProtocol);
     }
 
     /**
      * Get FileUploadNotificationReceiver object.
-     *
+     * @deprecated Use {@link #createFileUploadNotificationListener(FileUploadNotificationCallback)} instead
      * @return The instance of the FileUploadNotificationReceiver
      */
+    @Deprecated
     public FileUploadNotificationReceiver getFileUploadNotificationReceiver()
     {
-        return new FileUploadNotificationReceiver(hostName, userName, sasToken, iotHubServiceClientProtocol);
+        return new FileUploadNotificationReceiver(hostName, userName, makeSasToken(), iotHubServiceClientProtocol);
+    }
+
+    /**
+     * Create an instance of a file upload notification listener. This instance allows you to open a connection that will receive
+     * file upload notifications over time. Whenever a file upload notification is received, the provided callback will be executed,
+     * allowing you to complete/abandon/reject each notification. If called multiple times, multiple listener instances will be created.
```

Comment:
Is there any limit on how many instances can be created? We should also add a note explaining how callbacks are handled when we have multiple listeners attached to a single hub instance.

Reply:
There is no limit, so I'll add a note about that, sure